site/src/pages/WorkspacePage/WorkspaceReadyPage.tsx

```diff
@@ -100,13 +100,11 @@ export const WorkspaceReadyPage = ({
       ["canceling", "deleting", "pending", "starting", "stopping"].includes(
```

Comment:
You can't update the workspace to a template with a required, ephemeral parameter. In that case, you need to ask for the parameter value.

<img width="1504" alt="Screenshot 2023-07-17 at 13 37 32" src="https://github.com/coder/coder/assets/14044910/0bd4a9f5-5c8e-4b0e-a508-402fe93b4a7c">

Reply:
Do we already have an error like this to catch missing parameters:
```ts
 isMissingBuildParameterError: (_, { data }) => {
  return data instanceof API.MissingBuildParameters
},
```
I think we should use the same error for this. What do you think?

Reply:
I merged the PR making ephemeral parameters optional-only and merged this change into your branch. Feel free to re-pull it.

Reply:
Yes, we can resolve this thread 👍 .